### PR TITLE
Upgrade from istanbul to nyc

### PR DESCRIPTION
**Pull request by neilbmclaughlin**

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
Issues resolved.

**Comment by st3v3nhunt** [comment]
Rebase required please.

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,6 @@ newrelic_agent.log
 backstop-tests/backstop_data/html_report/
 backstop-tests/backstop_data/bitmaps_test/
 backstop-tests/.idea
+
+# nyc test coverage
+.nyc_output
```
Comment (st3v3nhunt) [inline .gitignore]: `test-results.xml` needs adding please.

Comment (neilbmclaughlin) [reply .gitignore]: Done

**Pull request merged**